timer_cli/auth.py

```diff
@@ -108,14 +116,20 @@ def default(x):
             return str(x)
 
         if self.modified:
-            with open(self.token_store, "w") as f:
-                json.dump(
-                    {k: v._asdict() for k, v in self.tokens.items()},
-                    f,
-                    indent=2,
-                    sort_keys=True,
-                    default=default,
-                )
+            # disable permissions other than user read/write
+            original_umask = os.umask(0o177)
```

Comment:
I'm not familiar with the octals - what does temporarily setting the umask to 0o177 do for the subsequent file operations?

Reply:
The `umask` setting _disables_ permissions for files created in a process, so a setting of 177 here means that the execute permission for users will be disabled and all permissions for anyone else are disabled.